src/AdminBuilder.js

```diff
@@ -1,37 +1,61 @@
-import React, {Component} from 'react';
-import PropTypes from 'prop-types';
-import {Admin, Resource, Delete} from 'admin-on-rest';
 import Api from '@api-platform/api-doc-parser/lib/Api';
-import List from './List';
-import Show from './Show';
+import {Admin, Delete, Resource} from 'admin-on-rest';
+import PropTypes from 'prop-types';
+import React, {Component} from 'react';
 import Create from './Create';
 import Edit from './Edit';
+import fieldFactory from './fieldFactory';
+import inputFactory from './inputFactory';
+import List from './List';
+import Show from './Show';
 
 export default class extends Component {
+  static defaultProps = {
+    fieldFactory,
+    inputFactory,
+  };
+
   static propTypes = {
     api: PropTypes.instanceOf(Api).isRequired,
+    fieldFactory: PropTypes.func,
+    inputFactory: PropTypes.func,
     restClient: PropTypes.func.isRequired,
   };
 
   render() {
-    let props = {...this.props};
+    const props = {...this.props};
     if (!props.title) props.title = this.props.api.title;
```

Comment:
Why? BTW sometimes you use this.props and sometimes props.
If you use a stateless component you won't have the choice but to keep consistency. 😋